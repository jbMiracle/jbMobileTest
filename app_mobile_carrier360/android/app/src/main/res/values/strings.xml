<resources>
    <string name="app_name">app_mobile_carrier360</string>
</resources>
