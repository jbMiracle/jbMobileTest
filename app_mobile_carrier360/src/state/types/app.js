const ActionNamespace = 'App';

export const Genesis = `${ActionNamespace}/GENESIS`;
export const ForgotPassword = `${ActionNamespace}/FORGOT_PASSWORD`;
export const ForgotPasswordDone = `${ForgotPassword}/DONE`;
