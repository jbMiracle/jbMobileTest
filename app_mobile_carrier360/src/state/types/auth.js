const ActionNamespace = 'Auth';

export const QueryBiometricSupport = `${ActionNamespace}/QUERY_BIOMETRIC_SUPPORT`;
export const BiometricSupportDetermined = `${ActionNamespace}/BIOMETRIC_SUPPORT_DETERMINED`;
export const SetupBiometricLogin = `${ActionNamespace}/SETUP_BIOMETRIC_LOGIN`;
export const BiometricLoginSetup = `${ActionNamespace}/BIOMETRIC_LOGIN_SETUP`;
// export const InitialLogin = `${ActionNamespace}/INITIAL_LOGIN`; TODO: Maybe use this for any Initial Login actions??

export const LoginRequested = `${ActionNamespace}/LOGIN_REQUESTED`;
export const LoginRejected = `${ActionNamespace}/LOGIN_REJECTED`;
export const LoginAccepted = `${ActionNamespace}/LOGIN_ACCEPTED`;
export const UserAuthenticated = `${ActionNamespace}/USER_AUTHENTICATED`;
export const Logout = `${ActionNamespace}/LOGOUT`;
