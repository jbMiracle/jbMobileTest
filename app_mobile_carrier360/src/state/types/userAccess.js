const ActionNamespace = 'USER_ACCESS';

const UserAccess = {
  LoginStart: `${ActionNamespace}/LOGIN/START`,
  LoginPayloadFetched: `${ActionNamespace}/LOGIN/PAYLOAD_FETCHED`,
  LoginDone: `${ActionNamespace}/LOGIN/DONE`,
  PostLogin: `${ActionNamespace}/SAGA/POST_LOGIN/DONE`,
  LogoutStart: `${ActionNamespace}/LOGOUT/START`,
  LogoutDone: `${ActionNamespace}/LOGOUT/DONE`,
};

export default UserAccess;
