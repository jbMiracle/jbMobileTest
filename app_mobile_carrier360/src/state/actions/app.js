import { App as AppActionTypes } from '../types';

export const genesis = () => ({
  type: AppActionTypes.Genesis,
});

export const forgotPassword = email => ({
  type: AppActionTypes.ForgotPassword,
  payload: email,
});

export const forgotPasswordServiceDone = (success, errorMessage) => ({
  type: AppActionTypes.ForgotPasswordDone,
  payload: {
    success,
    errorMessage,
  },
});
