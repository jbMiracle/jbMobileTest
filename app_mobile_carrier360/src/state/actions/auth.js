import { Auth as AuthActionTypes } from '../types';

export const userAuthenticated = (result = true) => ({
  type: AuthActionTypes.UserAuthenticated,
  payload: result,
});

export const queryBiometricSupport = () => ({
  type: AuthActionTypes.QueryBiometricSupport,
});

export const biometricSupportDetermined = supportResult => ({
  type: AuthActionTypes.BiometricSupportDetermined,
  payload: supportResult,
});

export const setupBiometricLogin = () => ({
  type: AuthActionTypes.SetupBiometricLogin,
});

export const biometricLoginSetup = (isSetup = true) => ({
  type: AuthActionTypes.BiometricLoginSetup,
  payload: { isSetup },
});
