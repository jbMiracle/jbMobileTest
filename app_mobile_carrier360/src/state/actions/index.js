import * as App from './app';
import * as Auth from './auth';
import * as UserAccess from './userAccess/UserAccessActions';

export {
  App,
  Auth,
  UserAccess,
};
