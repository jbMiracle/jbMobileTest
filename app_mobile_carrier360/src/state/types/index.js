import * as App from './app';
import * as Auth from './auth';
import UserAccess from './userAccess';

export {
  App,
  Auth,
  UserAccess,
};
