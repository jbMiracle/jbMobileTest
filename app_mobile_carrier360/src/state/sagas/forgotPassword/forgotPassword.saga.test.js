import SagaTester from 'redux-saga-tester';
import { App as AppActionTypes } from '../../types';
import { App as AppActions } from '../../actions';
import forgotPasswordSaga from '.';
import strings from '../../../services/localization/localization.service';
import sendForgotPassword from '../../../services/forgotPassword/forgotPassword.service';

jest.mock('../../../services/forgotPassword/forgotPassword.service', () => jest.fn());

describe('ForgotPasswordSaga', () => {
  let sagaTester = null;

  beforeEach(() => {
    sagaTester = new SagaTester({ });
    sagaTester.start(forgotPasswordSaga);
  });

  it('can successfully execute the forgotPassword saga without issues', async () => {
    sendForgotPassword.mockImplementation(() => Promise.resolve({}));

    sagaTester.dispatch(AppActions.forgotPassword('test@test.com'));
    await sagaTester.waitFor(AppActionTypes.ForgotPasswordDone);
    expect(sagaTester.getLatestCalledAction()).toEqual(AppActions.forgotPasswordServiceDone(true, null));
    expect(sagaTester.getState()).toMatchSnapshot();
  });

  it('will fail because the service had a fatal error', async () => {
    sendForgotPassword.mockImplementation(() => { throw Error(strings('ForgotPassword.errors.user_not_found')); });

    sagaTester.dispatch(AppActions.forgotPassword('test@test.com'));
    await sagaTester.waitFor(AppActionTypes.ForgotPasswordDone);
    expect(sagaTester.getLatestCalledAction()).toEqual(AppActions.forgotPasswordServiceDone(false, strings('ForgotPassword.errors.user_not_found')));
    expect(sagaTester.getState()).toMatchSnapshot();
  });

  it('will fail because the service had a non-fatal error', async () => {
    sendForgotPassword.mockImplementation(() => false);

    sagaTester.dispatch(AppActions.forgotPassword('test@test.com'));
    await sagaTester.waitFor(AppActionTypes.ForgotPasswordDone);
    expect(sagaTester.getLatestCalledAction()).toEqual(AppActions.forgotPasswordServiceDone(false, strings('App.genericError')));
    expect(sagaTester.getState()).toMatchSnapshot();
  });
});
