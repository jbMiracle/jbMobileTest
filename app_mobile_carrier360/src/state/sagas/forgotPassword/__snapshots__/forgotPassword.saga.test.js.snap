// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordSaga can successfully execute the forgotPassword saga without issues 1`] = `Object {}`;

exports[`ForgotPasswordSaga will fail because the service had a fatal error 1`] = `Object {}`;

exports[`ForgotPasswordSaga will fail because the service had a non-fatal error 1`] = `Object {}`;
